### PR TITLE
added RecordClient and entity

### PR DESCRIPTION
**Pull request by BenjiHartley**

### CODE REVIEW

File: src/DDoSX/RecordClient.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+     * @param int $page
+     * @param int $perPage
+     * @param array $filters
+     * @return int|\UKFast\SDK\Page
+     */
+    public function getPage($domainName, $page = 1, $perPage = 20, $filters = [])
+    {
+        $page = $this->paginatedRequest('v1/domains/' . $domainName . '/records', $page, $perPage, $filters);
+        $page->serializeWith(function ($item) {
+            return new Record($item);
```
Comment (Gman98ish) [inline src/DDoSX/RecordClient.php]: I think you'll need to map the API properties to the SDK properties.

I imagine the api doesn't return `safednsRecordId` but probably returns `safedns_record_id`

You can do this with

```php
return new Record($this->apiToFriendly($item, ['safedns_record_id' => 'safednsRecordId']));
```

You can do this with

File: src/DDoSX/RecordClient.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        $filters = $this->friendlyToApi($filters, $this->requestMap);
+
+        $page = $this->paginatedRequest('v1/domains/' . $domainName . '/records', $page, $perPage, $filters);
+        $page->serializeWith(function ($item) {
+            return new Record($this->apiToFriendly($item, $this->requestMap));
+        });
+
+        return $page;
+    }
+}
```
Comment (phily245) [inline src/DDoSX/RecordClient.php]: This will fix the failing CI:

```suggestion
}

```

File: src/DDoSX/RecordClient.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
+<?php
```
Comment (phily245) [inline src/DDoSX/RecordClient.php]: Please could we add the shortcut method onto `UKFast\SDK\DDoSX\Client`?

Comment (BenjiHartley) [reply src/DDoSX/RecordClient.php]: Added method

**Pull request merged**